fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/LogicalTypeFilterExpression.java

```diff
@@ -134,7 +161,47 @@ public Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candid
                                           @Nonnull final AliasMap aliasMap,
                                           @Nonnull final IdentityBiMap<Quantifier, PartialMatch> partialMatchMap,
                                           @Nonnull final EvaluationContext evaluationContext) {
-        return exactlySubsumedBy(candidateExpression, aliasMap, partialMatchMap);
+        final Iterable<MatchInfo> subsumedMatches = exactlySubsumedBy(candidateExpression, aliasMap, partialMatchMap);
```

Comment:
I would probably do this:

1. check if both are `LogicalTypeFilterExpression`s. If not return empty.
2. check if the candidate expression has a place holder, if not straight out call exactlySubsumedBy() and return that
3. all the special stuff when there is a placeholder